### DISCUSSION TITLE
Causing CPU to ramp up to 100% usage after about an hour of having the macro running.

### DISCUSSION THREAD

**Discussion by TheStonedSpider** [WarmUpTill/SceneSwitcher#1261] [General]
So I have about 6 or so scenes on a random play macro thru this obs addon and after about an hour, my cpu (ryzen 5600x) would ramp uo to 100% usage. Why this isn't using the hardware i.e. gfx card is beyond me. Each of these scenes show diff videos and all the videos are at 1080p and arent spectacular. just mp4s
I feel like it might not be shutting down previous scenes when switching to the next one. Ive noticed this when it isnt running that the scenes wont shut down when you switch scenes like it did before installing this addon. I have to go and manually shut everything down in the scene or itll keep playing. ive tried to find a setting that might be interfering with this but could not.

**Comment by WarmUpTill** [comment]
Hm ... that sounds very strange.
Can you share the macros you are using?
Preferably by [exporting them to a file](https://github.com/WarmUpTill/SceneSwitcher/wiki/Saving-and-loading-settings#how-to-create-and-import-a-backup-of-your-settings).

>I feel like it might not be shutting down previous scenes when switching to the next one.

The plugin should be not changing the behavior of how OBS handles scene switches internally.

Can you elaborate what you mean by "shutting down scenes"?

I just to make sure I don't misunderstand what the problem is.
I assume you are just referring to stopping the playback of media sources.

>ive tried to find a setting that might be interfering with this but could not.

If you are indeed only talking about media sources, then the behavior what happens, if the source becomes "invisible", can be configured for the VLC source at least in the source's settings.

![image](https://github.com/user-attachments/assets/a99de800-43fd-40cc-b66b-71ee8a3611b2)

(With "invisible" I mean the source is not part of the currently active program or preview scene or displayed in a "projector" window.)

Unfortunately, I don't think there is such an option for the "Media Source" source type.
It will continue playing until the playback ends.
But that is not something that is "changed" by the advanced scene switcher in any way.

You could however set up a macro to explicitly stop the playback of all media sources and VLC sources when the scene changed:

![image](https://github.com/user-attachments/assets/669370be-1182-42f9-9e8c-44df7f8f2959)

You just have to make sure to configure the sources to restart their playback once they become active again.

**Comment by TheStonedSpider** [comment]
Oh I did it all according to the tutorials and reading the wiki. Its working as its supposed to and I really dont know how I did anything without it. But i now realize the issue is with obs is decoding the videos. its just that this extension brought it to light cuz of the videos being part of the scenes in the macro. OBS for some reason on some files will max out the cpu for no reason. and it doesnt do it to the same video twice in a row. its always random and ive brough this up to obs numerous times and everytime they just dismissed it. been happening for a few years now regardless of PC, obs build, etc.
Also, what I mean by shutting down previous scenes is that the previous scene kept playing in the background even tho it was supposed to shut down. but i figured out that one as well after a few hours of making it reoccur. I accidently had a scene in the macro with nothing in it so when it hit that one the previous macro just kept playing and i thought it wasnt shutting down the previous scene properly,  i.e. pausing/stopping the video playback, browser sources etc.
I wasnt actually accusing the extension of making the cpu ramp up even tho the wording would seem that way. i was going to come back and give an update and use it as heads up for others and forgot to come back. thats my bad.
